api/lib/topic-lib.js

```diff
@@ -16,4 +16,50 @@ function addToTopicDB(options, callback){
   })
 }
 
-module.exports.addToTopic = addToTopicDB;
\ No newline at end of file
+module.exports.getAllTopics = function getAllTopics(options, callback){
+  Topic.find({}, function(err, topics) {
+    if(err)
+      return callback(err);
+    return callback(null, topics);
+  });
+}
+
+module.exports.addTopic = function addTopic(options, callback){
+  var newTopic = new Topic({
+    topicQuestion: options.question,
+    tag: options.tag,
+    authorName: options.author
+  });
+  newTopic.save(function (err) {
+    if (err) {
+      return callback(err);
+    }
+    // saved!
+    console.log('saved to topic');
+    return callback(null);
+  })
+}
+
+module.exports.getTopicById = function getTopicById(options, callback){
+  Comment.find({ topic: options._id }).
+  exec(function (err, comments) {
+    if(err)
```

Comment:
Error is not getting caught here. If the topic does not exist then it returns 500 error and body is `Bad String`